linkcheck/strformat.py

```diff
@@ -54,10 +56,10 @@ def unicode_safe (s, encoding=i18n.default_encoding, errors='replace'):
     @rtype: unicode
     """
     assert s is not None, "argument to unicode_safe was None"
-    if isinstance(s, unicode):
+    if isinstance(s, str_text):
         # s is already unicode, nothing to do
         return s
-    return unicode(str(s), encoding, errors)
+    return str_text(bytes(s), encoding, errors)
```

Comment:
This is the same code as before?  I'm confused now; how does this PR differ from #228?

Reply:
To be more clear, I expected #228 to be split into two PRs:
- one with two small changes to ascii_safe and unicode_safe 
- one with all the rest of the changes

and here I'm seeing all changes...

Wait is this one based on top of #228, and you force-pushed that one to drop the ascii_safe/unicode_safe bits?  Let me check....

Reply:
Oooh, you did!  Sorry, didn't realize, GitHub's PR review UI is rather suboptimal.

Reply:
(There's also a slight terminology confusion: where you said "branched from #NNN" I misread as "split off #NNN" rather than "based on top of #NNN so please merge that one first to see the actual diff".  Not your fault at all!)